src/MainProcessor.cpp

```diff
@@ -26,11 +35,47 @@ void CMainProcessor::process(float** ppfOutBuffer, int iNumChannels, int iNumFra
 }
 
 
+void CMainProcessor::checkFlags()
+{
+	CInstrument::checkFlags();
+}
+
 void CMainProcessor::checkTriggers()
 {
 	CScheduler::checkTriggers();
-	m_MapNoteOn.erase(m_iSampleCounter);
-	m_MapNoteOff.erase(m_iSampleCounter);
-	m_MapRemover.erase(m_iSampleCounter);
+
+	// Erases data entries that will never again be accessed
+	m_MapNoteOn.erase(m_iSampleCounter.load());
+	m_MapNoteOff.erase(m_iSampleCounter.load());
+	m_MapRemover.erase(m_iSampleCounter.load());
 }
 
+void CMainProcessor::checkQueues()
+{
+	// Places event and instrument pointer into appropriate container
+	std::pair<std::shared_ptr<CInstrument>, std::optional<TriggerInfo>> instToAdd;
+	while (m_InsertQueue.pop(instToAdd))
+	{
+		std::shared_ptr<CInstrument> pInstToAdd = instToAdd.first;
+		auto triggerInfo = instToAdd.second;
+		if (triggerInfo.has_value())
+		{
+			// Here m_iSampleCounter is added to trigger values so that they are relative to the CURRENT POSITION IN TIME
+			m_MapNoteOn[static_cast<int64_t>(triggerInfo.value().noteOn) + m_iSampleCounter.load()].insert(pInstToAdd);
+			m_MapNoteOff[static_cast<int64_t>(triggerInfo.value().noteOff) + m_iSampleCounter.load()].insert(pInstToAdd);
+			m_MapRemover[static_cast<int64_t>(triggerInfo.value().remove) + m_iSampleCounter.load()].insert(pInstToAdd);
+		}
+		else
+		{
+			m_SetInsts.insert(pInstToAdd);
+		}
+	}
+
+	std::shared_ptr<CInstrument> instToRemove = 0;
+	while (m_RemoveQueue.pop(instToRemove))
+	{
+		m_SetInsts.erase(instToRemove);
```

Comment:
I think you are right to be worried about `shared_ptr` causing deletes in the audio thread.
Check [this talk](https://www.youtube.com/watch?v=boPEO2auJj4&t=2394s) out, if you haven't seen it already - came across it in the JUCE forums, and it has a bit around 46:30 that addresses this exact problem. (The solution given there is to keep an extra reference around in another vector, and periodically handle deletions in e.g. the GUI thread.)

Anyway - for this PR, I think it's fine to add a TODO for now.